src/local.jl

```diff
@@ -149,6 +142,20 @@ function localzone()
     error("Failed to find local time zone")
 end
 
+# Extract a time zone name from a path.
+# e.g. "/usr/share/zoneinfo/Europe/Warsaw" becomes "Europe/Warsaw"
+function _path_tz_name(path::AbstractString, mask::Class=Class(:ALL))
+    i = 0
```

Comment:
This changes the behaviour as before we were starting off with i = 1 in the default case?

Reply:
Before this PR `i = findnext(isequal('/'), path, 1)`. The reason this was problematic was that for time zone names without a `/` we would fail to return a time zone. e.g. "UTC". (I'll note this also fails in other specific relative paths but that is harder to explain)